extensions/Vendor/EXT_texture_transform/README.md

```diff
@@ -0,0 +1,82 @@
+# EXT_texture_transform
+
+## Contributors
+
+* Steven Vergenz, Microsoft ([steven.vergenz@microsoft.com](mailto:steven.vergenz@microsoft.com))
+
+## Status
+
+Draft
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+Many techniques can be used to optimize resource usage for a 3d scene. Chief among them is the ability to minimize the number of textures the GPU must load. To achieve this, many engines encourage packing many objects' low-resolution textures into a single large texture atlas. The region of the resulting atlas that corresponds with each object is then defined by vertical and horizontal offsets, and the width and height of the region.
+
+To support this use case, this extension adds `offset` and `scale` properties to textureInfo structures. These properties would typically be implemented as transforms on the UV coordinates. In GLSL: `vec2 uvTransformed = Scale * Uv + Offset;`
+
+This is equivalent to Unity's `Material#SetTextureOffset` and `Material#SetTextureScale`, or Three.js's `Texture#offset` and `Texture#repeat`.
+
+## glTF Schema Updates
+
+The `EXT_texture_transform` extension may be defined on `textureInfo` structures. It may contain the following properties:
+
+| Name       | Type       | Default      | Description
+|------------|------------|--------------|---------------------------------
+| `offset`   | `array[2]` | `[0.0, 0.0]` | The offset of the UV coordinate origin as a percentage of the texture dimensions.
+| `scale`    | `array[2]` | `[1.0, 1.0]` | The scale factor applied to the components of the UV coordinates.
+| `texCoord` | `integer`  | `0`          | Overrides the textureInfo texCoord value if this extension is supported.
+
+Though this extension's values are unbounded, they will only produce sane results if the texture sampler's `wrap` mode is `REPEAT`, or if the result of the final UV transformation is within the range [0, 1] (i.e. negative scale settings and correspondingly positive offsets).
+
+> **Implementation Note**: For maximum compatibility, it is recommended that exporters generate UV coordinate sets both with and without transforms applied, use the post-transform set in the texture `texCoord` field, then the pre-transform set with this extension. This way, if the extension is not supported by the consuming engine, the model still renders correctly. Including both will increase the size of the model, so if including the fallback UV set is too burdensome, either add this extension to `extensionsRequired` or use the same texCoord value in both places.
```

Comment:
When implementing an exporter, will developers realistically follow this note? If I am generating UV coordinate sets with the transforms applied, is there any advantage to also including the extension?

Reply:
> if including the fallback UV set is too burdensome, ... or **use the same texCoord value in both places**

How will clients not supporting this extension render an asset correctly in such case?

Reply:
@lexaknyazev in this case, it would not be rendered "correctly", but in some cases a misplaced texture is preferable to a model failing to load entirely. It is up to creators/exporters to make this distinction.

Reply:
@donmccurdy to be honest, I don't really expect anyone to pre-multiply UV sets. But I thought it was worth noting how to work around lack of support explicitly.